Update jq to 1.4

jakirkham opened the pull request:
Updated the version of `jq`. Originally, this included `bison`. This has been dropped as patching `configure` allowed this error to be bypassed. As it was determined that `bison` was not being used, excluding this dependencies did not cause a problem.

ccordoba12 commented:
You also need `libtool` to build `jq`, at least according to its Readme

https://github.com/stedolan/jq/blob/master/README.md

ccordoba12 commented:
And it can also be built without `bison`, if you are not planning to modify it.

jakirkham commented:
> And it can also be built without bison, if you are not planning to modify it.

Unfortunately, that feature is not implemented until version 1.5 ( https://github.com/stedolan/jq/issues/689#issuecomment-74446461 ), which has not yet been released.

The only reason I included `bison` in this PR as opposed to the last one is the version from Apple's SDK is too old in this case and it fails to build if it doesn't find an acceptable version. On Linux, it looked for `bison`, which I didn't have, and proceeded on without it.

<br>
> You also need libtool to build jq, at least according to its Readme
> 
> https://github.com/stedolan/jq/blob/master/README.md

As far as `libtool` goes, on Mac the version included worked out of the box. On Linux, this was a bit weird, it appeared to somehow create `libtool` in the `jq` work directory, which I don't understand.

jakirkham commented:
I was able to patch the configure file based on this suggestion ( https://github.com/stedolan/jq/issues/689#issue-56239189 ). As a consequence, I have changed this PR to ignore the bison constraint. I created a separate PR for bison as it seems useful to have regardless ( https://github.com/conda/conda-recipes/pull/329 ). This has been retested and works on both OSes.

jakirkham commented:
@ccordoba12, I think I have addressed your concerns. This now works without `bison` and no longer includes the recipe. I have updated the summary to reflect this. Please let me know if you have any more suggestions.

ccordoba12 commented:
Thanks.
